app/services/github/profile_consumer.rb

```diff
@@ -1,4 +1,30 @@
 module Github
   class ProfileConsumer
+    attr_accessor :github_profile, :github_profile_response_code
+
+    GITHUB_RESPONSE_FIELDS = %w[login avatar_url html_url name company blog email bio twitter_username public_repos public_gists followers following updated_at location].freeze
+    PROFILE_FIELDS = %i[nickname avatar_url url name company blog email bio twitter_username public_repos public_gists followers following git_date location].freeze
+
+    def initialize(user_name)
+      github_response = ApiClient.new(user_name).call_for_github_profile_data
+      self.github_profile_response_code = github_response[:code]
+      self.github_profile = github_response[:body]
+    end
```

Comment:
IMO constructors should be as lightweight as possible. Here you should just set the `username` attribute and no more. The logic to call `ApiClient` to trigger the request and then save the response should be part of the `#call` method

```suggestion
    def initialize(username)
     self.username = username
    end
```

Reply:
If you follow my suggestion on setting the `code` and `body` as attributes for the `ApiClient` instance, this code (moved to the `#call` method would look like this:
```ruby
client = ApiClient.new(user_name)
client.call_for_github_profile_data

self.github_profile_response_code = client.code
self.github_profile = client.body
```

Reply:
Did